functions/uploads.py

```diff
@@ -300,7 +324,29 @@ def _get_bucket_and_blob(
         )
         return (bucket_name, make_pseudo_blob(object_name))
 
+    # get the storage client
     storage_client = get_storage_client()
-    bucket = storage_client.get_bucket(bucket_name)
+
+    # get the bucket and blob of interest.
+    # try to refresh 3 times if necessary
```

Comment:
If it fails on our first retry, is there reason to expect it would work on the second retry? I would suggest we keep this simpler and try refreshing the storage client just once until we see evidence that we need to retry multiple times.

Reply:
Also, there's still the call to `get_storage_client` in `_gcs_add_prefix_reader_permission`. Maybe this retry logic should be encapsulated in, e.g., a `_get_bucket_with_client_refresh` helper function, so both `_get_bucket_and_blob` and `_gcs_add_prefix_reader_permission` can share it.

Reply:
Sure, fair comments. No reason to expect something different.